edb/edgeql/compiler/setgen.py

```diff
@@ -785,6 +785,23 @@ def extend_path(
 
         src_path_id = source_set.path_id
 
+    orig_ptrcls = ptrcls
+
+    # For view ptrclses, find the place where the pointer is "really"
+    # defined that is, either its schema definition site or where it
+    # last had a expression defining it.
+    #
+    # This makes it so that views don't change path ids unless they are
+    # introducing some computation.
+    while (
+        ptrcls.get_is_derived(ctx.env.schema)
```

Comment:
Hmm, this reminds me of `get_nearest_non_derived_parent()`.  Perhaps this should be formulated as a method on `Pointer` as well?  `get_nearest_defined()` or something.